sdk/iot/hub/samples/src/paho_iot_hub_sas_telemetry_example.c

```diff
@@ -0,0 +1,418 @@
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      "Device ID", ENV_DEVICE_ID, "", false, device_id_span, &device_id_span));
+
+  az_span iot_hub_hostname_span = AZ_SPAN_FROM_BUFFER(iot_hub_hostname);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      "IoT Hub Hostname",
+      ENV_IOT_HUB_HOSTNAME,
+      "",
+      false,
+      iot_hub_hostname_span,
+      &iot_hub_hostname_span));
+
+  // Paho requires that the MQTT endpoint be of the form ssl://<HUB ENDPOINT>:8883
+  AZ_RETURN_IF_FAILED(
+      create_mqtt_endpoint(mqtt_endpoint, (int32_t)sizeof(mqtt_endpoint), iot_hub_hostname_span));
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      NULL));
+
+  return AZ_OK;
+}
+
+static int get_sas_key()
+{
+  az_result res;
+
+  // Create the POSIX expiration time from input hours
+  uint32_t sas_expiration = get_expiration_time(iot_hub_sas_key_expiration);
+
+  // Decode the base64 encoded SAS key to use for HMAC signing
+  az_span decoded_key_span;
+  if (az_failed(
+          res = sample_base64_decode(
+              iot_hub_sas_key_span, AZ_SPAN_FROM_BUFFER(sas_b64_decoded_key), &decoded_key_span)))
+  {
+    printf("Could not decode the SAS key: return code %d\n", res);
+    return res;
+  }
+
+  // Get the signature which will be signed with the decoded key
+  az_span sas_signature_span;
+  if (az_failed(
+          res = az_iot_hub_client_sas_get_signature(
+              &client,
+              sas_expiration,
+              AZ_SPAN_FROM_BUFFER(sas_signature_buf),
+              &sas_signature_span)))
+  {
+    printf("Could not get the signature for SAS key: return code %d\n", res);
+    return res;
+  }
+
+  // HMAC the signature with the decoded key
+  az_span encoded_span = AZ_SPAN_FROM_BUFFER(sas_signature_hmac_encoded_buf);
+  if (az_failed(
+          res
+          = sample_hmac_encrypt(decoded_key_span, sas_signature_span, encoded_span, &encoded_span)))
+  {
+    printf("Could not encrypt the signature: return code %d\n", res);
+    return res;
+  }
+
+  // base64 encode the result of the HMAC
+  az_span encoded_span_b64;
+  if(az_failed(res = sample_base64_encode(
+      encoded_span, AZ_SPAN_FROM_BUFFER(sas_signature_encoded_buf_b64), &encoded_span_b64)))
+  {
+    printf("Could not base64 encode the password: return code %d\n", res);
+    return res;
+  }
+
+  // Get the resulting password, passing the base64 encoded, HMAC signed bytes
+  size_t sas_key_length;
+  if (az_failed(
+          res = az_iot_hub_client_sas_get_password(
+              &client,
+              encoded_span_b64,
+              sas_expiration,
+              AZ_SPAN_NULL,
+              sas_key_password,
+              sizeof(sas_key_password),
```

Comment:
So the correct name according to the spec would be "sas_token", not sas_key_password.
We should either call it "mqtt_password" (what was advocated by @CIPop ) or "sas_token".

Reply:
Plus it matches the other variable `mqtt_username`, which is an easier mental model for customers to follow (people search for the combination of "username" and "password" when they are scanning code)

Reply:
done